### PR TITLE
Task_9952_9953_9954: Create UI, Handle Action and Load data Genres List Screen

### PR DESCRIPTION
**Pull request by khanhnn-1791**
### Issue Link :link:

https://edu-redmine.sun-asterisk.vn/issues/9952
https://edu-redmine.sun-asterisk.vn/issues/9953
https://edu-redmine.sun-asterisk.vn/issues/9954
### Goals :soccer:

https://marvelapp.com/project/3891276/screen/55150205
Create UI for Genres list screen
+ Back icon
+ Navigation Bar Title
+ Genres list using tableview
+ Genres item using tableviewcell include:
  - Genre icon
  - Genre title
Create a UI action
- Click genre item to go to detail genre including books
- Click back button to go to the previous view
Fetch Data from Local to display
### Implementation Details :construction:

Screen Shot 2019-04-12 at 1 19 10 PM

### CODE REVIEW

File: ios-i-book-lover/Scenes/Cell/GenreCell/GenreDisplayCell.swift [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import UIKit
+import Reusable
+
+final class GenreDisplayCell: UITableViewCell {
+    @IBOutlet private weak var genreImageView: UIImageView!
+    @IBOutlet private weak var genreLabel: UILabel!
+    
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        self.selectionStyle = .none
```
Comment (nguyenck-1-1519) [inline ios-i-book-lover/Scenes/Cell/GenreCell/GenreDisplayCell.swift]: self không cần thiết, self nên dùng khi assign biến cùng tên hoặc trong closure

File: ios-i-book-lover/Scenes/Cell/GenreCell/GenreDisplayCell.swift [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    }
+    
+    func configCell( _ genreTitle: String, _ genreImage: String) {
```
Comment (tgdong2296) [inline ios-i-book-lover/Scenes/Cell/GenreCell/GenreDisplayCell.swift]:
```suggestion
    func setContent( _ genreTitle: String, _ genreImage: String) {
```

File: ios-i-book-lover/Scenes/Cell/GenreCell/GenreDisplayCell.swift (hunk 1 above)
Comment (tgdong2296) [inline ios-i-book-lover/Scenes/Cell/GenreCell/GenreDisplayCell.swift]: ném vào 1 func configCell() r gọi trong này

File: ios-i-book-lover/Scenes/Home/GenresListViewController.swift [hunk 3]
```diff
@@ -10,11 +10,43 @@ import UIKit
 import Reusable
 
 final class GenresListViewController: UIViewController {
+    @IBOutlet private weak var tableView: UITableView!
+    var genresList = [String]()
```
Comment (tgdong2296) [inline ios-i-book-lover/Scenes/Home/GenresListViewController.swift:16]: break line

File: ios-i-book-lover/Scenes/Cell/GenreCell/GenreDisplayCell.swift (hunk 1 above)
Comment (tienpx3010) [inline ios-i-book-lover/Scenes/Cell/GenreCell/GenreDisplayCell.swift]:
```suggestion
        selectionStyle = .none
```

File: ios-i-book-lover/Scenes/Cell/GenreCell/GenreDisplayCell.swift (hunk 2 above)
Comment (tienpx3010) [inline ios-i-book-lover/Scenes/Cell/GenreCell/GenreDisplayCell.swift]:
```suggestion
    func configCell(genreTitle: String, genreImage: String) {
```

**Review by tgdong2296 — APPROVED** [review]

**Review by tienpx3010 — APPROVED** [review]

**Pull request merged**